File: API/Entities/ProductType.cs
namespace API.Entities;

public class ProductType
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string? PictureUrl { get; set; }
    public string? PublicId { get; set; }
    public List<Product> Products { get; set; } = new();

    public ProductType()
    {
    }

    public ProductType(string name, string? pictureUrl)
    {
        Name = name;
        PictureUrl = pictureUrl;
    }
}

File: client/src/app/models/productType.ts
import {ProductLite} from "./product";

export interface ProductType {
    id: number;
    name: string;
    pictureUrl?: string;
    publicId?: string;
}

export interface ProductTypeFull {
    id: number;
    name: string;
    pictureUrl?: string;
    products: ProductLite[]
}